engine/src/main/java/com/redhat/cloud/notifications/processors/webhooks/WebhookTypeProcessor.java

```diff
@@ -133,21 +151,68 @@ public NotificationHistory doHttpRequest(Notification item, HttpRequest<Buffer>
                 HttpRequestImpl<Buffer> reqImpl = (HttpRequestImpl<Buffer>) req.getDelegate();
 
                 boolean serverError = false;
+                boolean isEmailEndpoint = item.getEndpoint().getType() == EMAIL_SUBSCRIPTION;
+                boolean shouldResetEndpointServerErrors = false;
                 if (resp.statusCode() >= 200 && resp.statusCode() < 300) {
                     // Accepted
                     Log.debugf("Webhook request to %s was successful: %d", reqImpl.host(), resp.statusCode());
                     history.setInvocationResult(true);
+                    shouldResetEndpointServerErrors = true;
                 } else if (resp.statusCode() >= 500) {
                     // Temporary error, allow retry
                     serverError = true;
                     Log.debugf("Webhook request to %s failed: %d %s", reqImpl.host(), resp.statusCode(), resp.statusMessage());
                     history.setInvocationResult(false);
+                    if (featureFlipper.isDisableWebhookEndpointsOnFailure()) {
+                        if (!isEmailEndpoint) {
+                            /*
+                             * The target endpoint returned a 5xx status. That kind of error happens in case of remote
+                             * server failure, which is usually something temporary. Sending another notification to
+                             * the same endpoint may work in the future, so the endpoint is only disabled if the max
+                             * number of endpoint failures allowed from the configuration is exceeded.
+                             */
+                            boolean disabled = endpointRepository.incrementEndpointServerErrors(item.getEndpoint().getId(), maxServerErrors);
+                            if (disabled) {
+                                disabledEndpointsServerErrorCount.increment();
+                                Log.infof("Endpoint %s was disabled because we received too many 5xx status while calling it", item.getEndpoint().getId());
+                                // TODO NOTIF-512 Send a notification to the org admin explaining the situation.
```

Comment:
We could use the internal event type - and configure the recipients to force send to admins.

Something else we could explore in the future is the use of default settings. i.e. if nothing is configured, send to all admins.

Reply:
> We could use the internal event type - and configure the recipients to force send to admins.

That's exactly my plan 😃